### PR TITLE
refactor: use flag api to load the config in nfs module

### PR DESCRIPTION
**Pull request by Smityz**
Use `flags.h` to load the config

### CODE REVIEW

File: src/dist/nfs/nfs_client_impl.h [hunk 1]
```diff
@@ -66,49 +66,44 @@ struct nfs_opts
 
     void init()
     {
-        nfs_copy_block_bytes =
-            (uint32_t)dsn_config_get_value_uint64("nfs",
-                                                  "nfs_copy_block_bytes",
-                                                  4 * 1024 * 1024,
-                                                  "max block size (bytes) for each network copy");
-        max_concurrent_remote_copy_requests = (int)dsn_config_get_value_uint64(
+        DSN_DEFINE_uint32("nfs",
```
Comment (levy5307) [inline src/dist/nfs/nfs_client_impl.h:69]: `nfs_copy_block_bytes` is not initialized in your code

File: src/dist/nfs/nfs_client_impl.h [hunk 2]
```diff
@@ -66,49 +66,55 @@ struct nfs_opts
...
```
Comment (neverchanje) [inline src/dist/nfs/nfs_client_impl.h:69]: Define flags in global space:

```
using TokenBucket = folly::BasicTokenBucket<std::chrono::steady_clock>;

DSN_DEFINE_uint32("nfs", xxx);
DSN_DEFINE_uint32("nfs", yyy);
DSN_DEFINE_uint32("nfs", zzz);



struct nfs_opts {
```

File: src/dist/nfs/nfs_client_impl.cpp [hunk 3]
```diff
@@ -40,7 +40,57 @@
...
+                 "maximum retry count when copy failed");
+DSN_DEFINE_int32("nfs",
+                 rpc_timeout_ms,
+                 10000,
+                 "rpc timeout in milliseconds for nfs copy, "
+                 "0 means use default timeout of rpc engine");
+
+void nfs_opts::init()
+{
+    nfs_copy_block_bytes = FLAGS_nfs_copy_block_bytes;
```
Comment (acelyc111) [inline src/dist/nfs/nfs_client_impl.cpp:83]: You can use FLAGS_nfs_copy_block_bytes directly, nfs_copy_block_bytes can be removed.
Another profit of using FLAGS_xxx is to support dynamic configs modification in the future.

File: src/dist/nfs/nfs_client_impl.cpp [hunk 4]
```diff
@@ -40,15 +40,50 @@
 namespace dsn {
 namespace service {
 
+DSN_DEFINE_uint32("nfs",
```
Comment (neverchanje) [inline src/dist/nfs/nfs_client_impl.cpp:43]: Why the original config is uint64 while you define here a uint32?

File: src/dist/nfs/nfs_server.h [hunk 5]
```diff
@@ -27,11 +27,16 @@
...
 #include <iostream>
 #include <dsn/cpp/serverlet.h>
+#include <dsn/utility/flags.h>
 #include "nfs_code_definition.h"
 #include "nfs_types.h"
 
 namespace dsn {
 namespace service {
+
+DSN_DECLARE_int32(file_close_timer_interval_ms_on_server);
```
Comment (acelyc111) [inline src/dist/nfs/nfs_server.h:37]: Move to cpp file

File: src/dist/nfs/nfs_client_impl.cpp [hunk 6]
```diff
@@ -162,7 +197,7 @@ void nfs_client_impl::end_get_file_size(::dsn::error_code err,
                 break;
             }
 
-            req_size = size > _opts.nfs_copy_block_bytes ? _opts.nfs_copy_block_bytes
+            req_size = size > FLAGS_nfs_copy_block_bytes ? FLAGS_nfs_copy_block_bytes
```
Comment (levy5307) [inline src/dist/nfs/nfs_client_impl.cpp:200]: std::min(static_cast<uint32_t>(size), FLAGS_nfs_copy_block_bytes);

**Pull request merged**